.github/workflows/benchmark.main.kts

```diff
@@ -0,0 +1,161 @@
+                    "include" to listOf(
+                        mapOf("os" to "ubuntu-latest"),
+                        mapOf(
+                            "os" to "macos-latest",
+                            "additional-args" to "-x jvmBenchmark -x jsBenchmark",
+                        ),
+                        mapOf(
+                            "os" to "macos-13", // for macosX64
+                            "additional-args" to "-x jvmBenchmark -x jsBenchmark",
+                        ),
+                        mapOf(
+                            "os" to "windows-latest",
+                            "additional-args" to "-x jvmBenchmark -x jsBenchmark",
+                        ),
+                    )
+                )
+            )
+        )
+    ) {
+        uses(action = Checkout())
+        uses(
+            name = "Set up JDK",
+            action = SetupJava(
+                javaVersion = "11",
+                distribution = SetupJava.Distribution.Zulu,
+                cache = SetupJava.BuildPlatform.Gradle,
+            ),
+        )
+        uses(
+            name = "Validate Gradle Wrapper",
+            action = ActionsWrapperValidation(),
+        )
+        uses(
+            name = "Setup Gradle",
+            action = ActionsSetupGradle(
+                gradleVersion = "wrapper",
+            ),
+        )
+        run(
+            name = "Run benchmarks",
+            command = "./gradlew -p snake-kmp-benchmarks benchmark --no-parallel ${ expr{ "matrix.additional-args" }}",
+        )
+        uses(
+            action = UploadArtifact(
+                name = "bench-results-${ expr { "matrix.os" } }",
+                path = listOf("$BENCHMARK_RESULTS/main/**/*.json"),
+            )
+        )
+    }
+
+    job(
+        id = "collect-benchmarks-results",
+        runsOn = RunnerType.UbuntuLatest,
+        needs = listOf(runBenchmark),
+    ) {
+        // without checkout step 'benchmark-action/github-action-benchmark' action won't work
+        uses(action = Checkout())
+        uses(
+            name = "Download benchmark results",
+            action = DownloadArtifact(
+                pattern = "bench-results-*",
+                path = RESULTS_DIR,
+                mergeMultiple = true,
+            )
+        )
+        @OptIn(ExperimentalKotlinLogicStep::class)
+        run(
+            name = "Prepare and join benchmark reports",
+        ) {
+            val mergedReports = File(RESULTS_DIR).walk()
+                .filter { it.extension == "json" }
+                .flatMap {
+                    val reportForPlatform = it.readText()
+                        // Trim lengthy Kotlin package and data file path to make benchmark name more readable
+                        .replace("it.krzeminski.snakeyaml.engine.kmp.benchmark", it.nameWithoutExtension)
+                        .replace(Regex("\"[^\"]+data"), "\"data")
+                    Json.parseToJsonElement(reportForPlatform).jsonArray
```

Comment:
note: In my opinion, it would be better (and safer) to transform the `JsonElement` rather then replacing values in the file with regex. But if the result is the same, I have no objections to this approach.

Reply:
You're right, I thought of it, but then realized that the values to be replaced sit three levels down in the JSON hierarchy, so it would make this code pretty complicated. That's why I went with this simpler approach, even though it may be a bit unsafe. Happy to revisit later if it turns out to be problematic.